aten/src/ATen/cudnn/Descriptors.h

```diff
@@ -227,14 +227,14 @@ struct TORCH_CUDA_CPP_API DropoutDescriptor
     AT_ASSERT(options.device().type() == kCUDA);
     AT_ASSERT(options.dtype() == kByte);
     state = at::empty({static_cast<int64_t>(state_size)}, options);
-    AT_CUDNN_CHECK(cudnnSetDropoutDescriptor(mut_desc(), handle, dropout, state.data_ptr(), state_size, seed));
+    AT_CUDNN_CHECK(cudnnSetDropoutDescriptor(mut_desc(), handle, dropout, state.mutable_data_ptr(), state_size, seed));
   }
 
   // Restore a dropout descriptor given a dropout probability and existing RNG state.
   void set(cudnnHandle_t handle, float dropout, at::Tensor state_) {
     TORCH_INTERNAL_ASSERT(dropout > 0, "dropout must be nonzero; otherwise call set_no_dropout");
     state = state_;
-    void *state_ptr = state.data_ptr();
+    void *state_ptr = state.mutable_data_ptr();
```

Comment:
Not sure about this one as dropout state may get subsequently mutated through the descriptor. You will need to read docs